scalameter-core/src/main/scala/org/scalameter/MeasureBuilder.scala

```diff
@@ -8,21 +8,26 @@ package org.scalameter
 class MeasureBuilder[T, U](
   val ctx: Context,
   val warmer: Warmer,
-  val measurer: Measurer,
+  val measurer: Measurer[U],
   val regen: () => T,
   val setup: T => Unit,
   val teardown: T => Unit,
-  val resultFunction: Seq[Double] => U
+  val resultFunction: Seq[U] => U
```

Comment:
It seems to me that we actually might want to have a `Seq[Quantity[U]]` here. Otherwise, there is no way to take the unit into account when writing the `resultFunction`. It's true that the workaround would be to use a `map` in the measurer to set the unit appropriately, but it would be more flexible to make the `resultFunction` take the full `Quantity`.